core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -289,27 +288,46 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
     }
   }
 
-  protected def setAclsAndProduce(tp: TopicPartition) {
+  protected def setAcls(tp: TopicPartition) {
     AclCommand.main(produceAclArgs(tp.topic))
     AclCommand.main(consumeAclArgs(tp.topic))
     servers.foreach { s =>
       TestUtils.waitAndVerifyAcls(TopicReadAcl ++ TopicWriteAcl ++ TopicDescribeAcl ++ TopicCreateAcl, s.dataPlaneRequestProcessor.authorizer.get,
         new Resource(Topic, tp.topic, PatternType.LITERAL))
       TestUtils.waitAndVerifyAcls(GroupReadAcl, s.dataPlaneRequestProcessor.authorizer.get, groupResource)
     }
+  }
+
+  protected def setAclsAndProduce(tp: TopicPartition) {
+    setAcls(tp)
     val producer = createProducer()
     sendRecords(producer, numRecords, tp)
   }
 
   /**
     * Tests that a producer fails to publish messages when the appropriate ACL
-    * isn't set.
+    * isn't set. Also verifies that subsequent send to authorized topic succeeds.
     */
-  @Test(expected = classOf[TopicAuthorizationException])
+  @Test
```

Comment:
Are the consumer and adminclient cases tested already?

Reply:
@ijuma Thanks for the review. This test verifies producer and consumer. AdminClient doesn't cache topics in its metadata, so that should be ok.

Reply:
The test comment only mentions the producer, so should probably be updated. Since this class is about end to end testing, I don't think it should care about implementation details. That is, we should have at least one test for the admin client where a batch of topics is partially authorized.